### DISCUSSION TITLE
Is there some interest in having an GroundWaterML2 import/export in flopy?

### DISCUSSION THREAD

**Discussion by rabbl** [modflowpy/flopy#2406] [Ideas]
[GroundWaterML2 (GWML2)](https://docs.ogc.org/is/19-013/19-013.html#toc64) is an Open Geospatial Consortium (OGC) standard designed to facilitate the exchange of groundwater model data within institutions and modelling frameworks. 

As it's approval is yet some years old, there is not much adoption in the world of groundwater modelling, as far we could do our research. 

As we are looking for funds for the standard implementation in Python and JS/TS, we are gathering the most useful use cases. So we ask you: Is there a basic interest within the flopy developer und user community to have GroundWaterML2 as Import/Export format plugin in place? 

Why do we think this idea is valid and fundable:

European public institutions are trying to switch to modelling approaches which can deliver the model data in a format which is standardised after finishing the work. With an open source solution like modflow/flopy we could develop this implementing an GWML-export/import and make this software unique in the market. Solution partners could be exchanged over the lifetime of a model. University and community driven modelling of public models becomes viable. 
With this unique feature, closed source solutions will have to follow this approach and closed models can be exported, validated, approved and also reused from other parties, when exported to the GWM2 format. This has a big advantage in our opinion to enhance the use of modflow and other open source solutions. 

The fund we are focussing on has a deadline which ends 2nd of January 2025. We are two developers which can spend 6 month on this issue. If you see also a valid use case for this feature we will put all our energy and effort to write a very good proposal before the deadline and submit it. If approved we can start full time work on this in mid to end 2025.

**Comment by christianlangevin** [comment]
Hi @rabbl, thanks for your interest in this.  I can give you my opinion regarding your proposal.  I think your ideas are good, and certainly having a way to export data into a standard format is appealing.  I wonder though about GWML2 and if it will become more widely adopted for groundwater modeling.  A tool will likely help with that.  I think from the flopy side, we would be willing to add lightweight hooks for such an import/export plugin.  I'm not sure that we would initially want to support large amounts of GWML2 export code inside flopy, as that would be an additional burden on us.  But we do have the concept of optional dependencies, and if you had a separate plugin package that could do most of the work, then we could call the optional dependency from our export/import routines.  That would allow you to maintain the GWML2 functionality, while making it accessible from flopy.  Would be good to know if anyone else has any thoughts.

**Reply by rabbl** [reply]
Thank you @langevin-usgs for sharing your opinion on this topic. 
We also thought about developing a plugin for flopy in the first place and see how it will be adopted from the community, research and environmental engineering. 

It's also not certain whether the standard can map all features from Flopy. We think this is an iterative process which once initiated will bring a lot of answers within the process itself. After this initial phase we will see if this standard can be used in the actual state or if relevant changes are required for further versions of the standard. 

We really hope to see further opinions from the flopy and modflow community which can help to bring this project to life.
